publishers/HumanRobotPosePublisher/HumanRobotPosePublisher.cpp

```diff
@@ -10,36 +10,90 @@
 
 #include <iDynTree/Core/Transform.h>
 
-#include <yarp/os/ResourceFinder.h>
+#include <iDynTree/yarp/YARPConversions.h>
 #include <yarp/dev/IFrameTransform.h>
 #include <yarp/dev/PolyDriver.h>
 #include <yarp/math/Math.h>
 #include <yarp/os/LogStream.h>
 #include <yarp/os/Node.h>
-#include <iDynTree/yarp/YARPConversions.h>
+#include <yarp/os/PortReader.h>
+#include <yarp/os/ResourceFinder.h>
+#include <yarp/os/RpcServer.h>
 
-#include <string>
+#include <atomic>
 #include <mutex>
+#include <string>
 
 const std::string DeviceName = "HumanRobotPosePublisher";
 const std::string LogPrefix = DeviceName + " :";
 constexpr double DefaultPeriod = 0.01;
 
 using namespace hde::publishers;
 
-class HumanRobotPosePublisher::impl
+class CmdParser : public yarp::os::PortReader
 {
+
 public:
+    std::atomic<bool> cmdStatus{false};
+    std::atomic<bool> resetStatus{false};
+
+    bool read(yarp::os::ConnectionReader& connection) override
+    {
+        yarp::os::Bottle command, response;
+        if (command.read(connection)) {
+
+            if (command.get(0).asString() == "help") {
+                response.addString("Enter <set> to set the robot pose correctly");
+            }
+            else if (command.get(0).asString() == "set") {
+                response.addString("Entered command <set> is correct");
+                this->cmdStatus = true;
+            }
+            else if (command.get(0).asString() == "reset") {
+                response.addString("Entered command <reset> is correct");
+                this->resetStatus = true;
+            }
+            else {
+                response.addString(
+                    "Entered command is incorrect, Enter help to know available commands");
+            }
+        }
+        else {
+            this->cmdStatus = false;
+            this->resetStatus = false;
+            return false;
+        }
+
+        yarp::os::ConnectionWriter* reply = connection.getWriter();
+
+        if (reply != NULL) {
+            response.write(*reply);
+        }
+        else
+            return false;
 
+        return true;
+    }
+};
+
+class HumanRobotPosePublisher::impl
+{
+public:
     mutable std::mutex mutex;
     bool first_data = false;
 
+    // Timeout
+    double tfTimeoutCheckDuration;
+
     yarp::dev::PolyDriver transformClientDevice;
     yarp::dev::IFrameTransform* iFrameTransform = nullptr;
 
+    // Final ground to robot base transform
+    yarp::sig::Matrix ground_H_robotBase;
+
     // Human Robot fixed transform
-    iDynTree::Transform humanRobotFixedTransform;
-    yarp::sig::Matrix robotLeftFoot_H_humanLeftFoot;
+    iDynTree::Transform robotHumanFixedTransform;
+    yarp::sig::Matrix humanLeftFoot_H_robotLeftFoot;
```

Comment:
Just to double check, is it correct left to left foot? If human and robot are facing, right to left seems more intuitive. I do not remember what we were using in the first place. If this is what you expect, ignore this comment.

Reply:
> If human and robot are facing, right to left seems more intuitive.

True and that is why i first tried to use the variable name `robotLeftFoot_H_humanLeftFoot` but on checking with the transformation used in HDEv1, I rolled back to using left to left feet transformation as it was double checked correctly.